src/test_l1_dsAudio.c

```diff
@@ -949,40 +1029,58 @@ void test_l1_dsAudio_positive_dsSetDialogEnhancement(void) {
-	intptr_t handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
+        dsError_t result;
+        int min_de_level = 0, max_de_level = 16, mid_de_level = 8;
+        intptr_t* handle = NULL;
 
-	// Step 01: Initialize audio ports
-	result = dsAudioPortInit();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // allocate memory for the handles
+        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+        if(handle == NULL)
+        {
+                UT_LOG_ERROR("Failed to allcoate memory for handlers");
+                return;
+        }
 
-	// Step 02: Get the port handle for all supported audio ports
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+        // Step 01: Initialize audio ports
+        result = dsAudioPortInit();
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-		// Step 03: Set Dialog Enhancement level - Min DE Level(0)
-		result = dsSetDialogEnhancement(handle[i], min_de_level);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // Step 02: Get the port handle for all supported audio ports
+        for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+                if(gDSAudioPortConfiguration[i].ms12_capabilites & 0x04)
+                {
+                        result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
+                        UT_ASSERT_EQUAL(result, dsERR_NONE);
+                        UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
-		// Step 04: Set Dialog Enhancement level - Max DE Level(16)
-		result = dsSetDialogEnhancement(handle[i], max_de_level);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
+                        // Step 03: Set Dialog Enhancement level - Min DE Level(0)
+                        result = dsSetDialogEnhancement(*(handle+i), min_de_level);
+                        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-		// Step 05: Set Dialog Enhancement level - Mid DE Level(16)
-		result = dsSetDialogEnhancement(handle[i], mid_de_level);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
+                        // Step 04: Set Dialog Enhancement level - Max DE Level(16)
+                        result = dsSetDialogEnhancement(*(handle+i), max_de_level);
+                        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-	}
+                        // Step 05: Set Dialog Enhancement level - Mid DE Level(16)
+                        result = dsSetDialogEnhancement(*(handle+i), mid_de_level);
+                        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-	// Step 06: Terminate audio ports
-	result = dsAudioPortTerm();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+                } else {
+                        result = dsSetDialogEnhancement(*(handle+i), max_de_level);
+                        UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
```

Comment:
check for NOT_SUPPORTED, not INVALID_PARAM
If capability not supported check for NOT_SUPPORTED. Check in other places as well

Reply:
done